### PR TITLE
Image Segmentation with an Improved UNet Model - Thomas Petersen - 45872093

### PR DESCRIPTION
**Pull request by Tompnyx**
Hi Shakes and helpers,
This pull request includes my implementation of the first problem presented in the assignment specification. It is an implementation of an improved UNet model that creates masks that segment lesions. This pull request consists of:

- `driver.py`: The driver script for my implementation. This will contain the logic for data pre-processing, model training/ validation/ testing, visual plotting, and model saving.
- `model.py`: The model script for my implementation. This contains the layer logic for my improved UNet model.
- `license.txt`: The Creative Commons license for my implementation.
- `readme.md`: This includes a quick report of my model: Explanation, examples, justifications, and references.
- `Example\model.png`: The visual representation of my improved UNet model's architecture. Used in the readme.md file.
- `Example\ModelLossPerformance.png`: A graph of an example of my model's training and validation loss over epochs run. Used in the readme.md file.
- `Example\ModelPerformance.png`: A graph of an example of my model's training and validation accuracy over epochs run. Used in the readme.md file.
- `Example\PredictedExample.png`: A plot of the original image, the true mask, and the predicted mask generated from my model. Used in the readme.md file.

I would greatly appreciate if you could review my pull request and provide any feedback.
Kind Regards,
Thomas (Tompnyx)

### CODE REVIEW

**Comment by wooboyeong** [comment]
General Comments
* You can threshold the output so that the mask is binary
* Code OK for EASY
* Commit messages OK
* Pull request OK

**Comment by shakes76** [comment]
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage 
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description good
